### PR TITLE
WebGLRenderer: Add "target" argument to "getClearColor"

### PR DESCRIPTION
**Pull request by gkjohnson**
Related issue: #20819

**Description**

When scene.background is set to a color the renderer clear color is modified because `WebGLRenderer.getClearColor()` returns a reference to its color instance.

Repro (can be run in console on docs page):

```js
renderer = new THREE.WebGLRenderer();
scene = new THREE.Scene();
scene.background = new THREE.Color( 0xff0000 );

console.log( renderer.getClearColor() );
generator = new THREE.PMREMGenerator( renderer );
generator.fromScene( scene );
console.log( renderer.getClearColor() );
```

In `dev` this logs `0, 0, 0` then `1, 0, 0`. With this PR both log statements print `0, 0, 0` as expected.

### CODE REVIEW

**Comment by Mugen87** [comment]
The fix is correct since the same code style is used in various post processing passes, too.

@mrdoob What do you think about rewriting `getClearColor()` similar to `getViewport()` or `getScissor()`.
```js
this.getClearColor = function ( target ) {

    if ( target === undefined ) {

        console.warn( 'WebGLRenderer: .getClearColor() now requires a Color as an argument' );

        target = new Color();

    }

    return target.copy( background.getClearColor() );

};
```
However, this style requires minor refactoring in the code base, too.

**Review by Mugen87 — APPROVED** [review]

**Comment by mrdoob** [comment]
> @mrdoob What do you think about rewriting `getClearColor()` similar to `getViewport()` or `getScissor()`.

Sounds good to me!

Would you like @gkjohnson to do that in this PR? Or should we leave it for a different PR?

**Comment by Mugen87** [comment]
> Or should we leave it for a different PR?

This one. In this way, the code base can migrated in one go.

**Comment by gkjohnson** [comment]
Done! Updated WebGLRenderer, docs, and examples.

I didn't touch WebGPURenderer but maybe that should be updated, too?

File: examples/js/postprocessing/ClearPass.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ THREE.ClearPass = function ( clearColor, clearAlpha ) {
 
 	this.clearColor = ( clearColor !== undefined ) ? clearColor : 0x000000;
 	this.clearAlpha = ( clearAlpha !== undefined ) ? clearAlpha : 0;
+	this.oldClearColor = new THREE.Color();
```
Comment (mrdoob) [inline examples/js/postprocessing/ClearPass.js:9]: Maybe `this._oldClearColor`?

**Comment by mrdoob** [comment]
Damn, this escalated quickly 😁

File: examples/js/postprocessing/ClearPass.js (hunk 1 above)
Comment (gkjohnson) [reply examples/js/postprocessing/ClearPass.js]: Done!

**Comment by gkjohnson** [comment]
> Damn, this escalated quickly

Heh, it's what you wanted, right?! 😁

**Comment by mrdoob** [comment]
Thanks!

**Pull request merged**